readthedocs/donate/forms.py

```diff
@@ -61,6 +61,27 @@ class Meta:
     last_4_digits = forms.CharField(widget=forms.HiddenInput(), required=True)
     name = forms.CharField(required=True)
     email = forms.CharField(required=True)
+    select_databind = "value: dollars_select, visible: dollars_select() != 'custom'"
+    select_widget = forms.Select(attrs={"data-bind": select_databind})
+    amounts = [
+        ('custom', "Custom amount"),
+        (5, "$5"),
+        (10, "$10"),
+        (25, "$25"),
+        (50, "1 Hour ($50)"),
+        (100, "2 Hours ($100)"),
+        (200, "4 Hours ($200)"),
+        (400, "1 Day ($400)"),
+        (800, "2 Days ($800)"),
+        (1200, "3 Days ($1200)"),
+        (1600, "4 Days ($1600)"),
+        (2000, "5 Days ($2000)"),
+        (4000, "2 Weeks ($4000)"),
+        (6000, "3 Weeks ($6000)"),
+        (8000, "4 Weeks ($8000)"),
+    ]
+    dollars_select = forms.ChoiceField(choices=amounts, initial=50,
+                                       widget=select_widget, label="Amount")
```

Comment:
A couple style notes here:

* Stylistically, we support either `amounts` as a class-level constant (`SupporterForm.DOLLAR_AMOUNTS`), or in a `constants` module (`readthedocs.donate.constants.DOLLAR_AMOUNTS`).
* `select_databind` and `select_widget` shouldn't be defined as variables, and should be defined in the same spot as the other bindings, in `Meta`
* Translation was lost on these strings